### PR TITLE
ethclient: added createAccessList

### PR DESCRIPTION
**Pull request by MariusVanDerWijden**
This PR adds the `eth_createAccessList` rpc call to the ethclient.

### CODE REVIEW

**Review by holiman — APPROVED** [review]
LGTM

File: ethclient/ethclient.go [hunk 1]
```diff
@@ -515,6 +515,21 @@ func (ec *Client) EstimateGas(ctx context.Context, msg ethereum.CallMsg) (uint64
 	return uint64(hex), nil
 }
 
+// CreateAccessList tries to create an access list for a specific transaction based on the
+// current pending state of the blockchain.
+func (ec *Client) CreateAccessList(ctx context.Context, msg ethereum.CallMsg) (*types.AccessList, uint64, string, error) {
```
Comment (rjl493456442) [inline ethclient/ethclient.go:520]: ethclient is designed for all eth-compatible clients. Is it true that `eth_createAccessList ` RPC endpoint is supported by all the (major) clients?

**Comment by fjl** [comment]
We have decided in triage that this is another good candidate for the "ethclient/gethclient" package.

**Comment by MariusVanDerWijden** [comment]
superseeded by https://github.com/ethereum/go-ethereum/pull/22977

**Pull request closed without merging**